include/dao/dao.js

```diff
@@ -710,6 +761,24 @@ module.exports = function DAOModule(pb) {
     };
 
     /**
+     * Gets all collection names
+     * @returns {DAO}
+     */
+    DAO.prototype.getAllCollections = function(cb) {
+        pb.dbm.getDb(this.dbName, function(err, db) {
+            if (util.isError(err)) {
+                return cb(err);
+            }
+            db.collections(function(err, items) {
```

Comment:
This is fixed.

Ben Tidwell
Software Engineer III | Talent Network
CareerBuilder
5550-A Peachtree Pkwy, Norcross GA 30092

From: Brian <notifications@github.com<mailto:notifications@github.com>>
Reply-To: pencilblue/pencilblue <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Sunday, July 12, 2015 at 6:57 PM
To: pencilblue/pencilblue <pencilblue@noreply.github.com<mailto:pencilblue@noreply.github.com>>
Cc: Ben Tidwell <ben.tidwell@careerbuilder.com<mailto:ben.tidwell@careerbuilder.com>>
Subject: Re: [pencilblue] Multi Tenancy (#680)


In include/dao/dao.js<https://github.com/pencilblue/pencilblue/pull/680#discussion_r34426700>:

> @@ -710,6 +761,24 @@ module.exports = function DAOModule(pb) {
>      };
>
>      /**
> +     * Gets all collection names
> +     * @returns {DAO}
> +     */
> +    DAO.prototype.getAllCollections = function(cb) {
> +        pb.dbm.getDb(this.dbName, function(err, db) {
> +            if (util.isError(err)) {
> +                return cb(err);
> +            }
> +            db.collections(function(err, items) {


The new mongo driver no longer supports calling collections.
http://mongodb.github.io/node-mongodb-native/2.0/api/Db.html#listCollections

Is this function really needed?

-
Reply to this email directly or view it on GitHub<https://github.com/pencilblue/pencilblue/pull/680/files#r34426700>.